src/journal.rs

```diff
@@ -0,0 +1,246 @@
+    }
+
+    fn sstore(
+        &mut self,
+        address: Address,
+        key: U256,
+        value: U256,
+    ) -> Result<StateLoad<SStoreResult>, <Self::Database as Database>::Error> {
+        JournalTr::sstore(&mut self.inner, address, key, value)
+    }
+
+    fn tload(&mut self, address: Address, key: U256) -> U256 {
+        JournalTr::tload(&mut self.inner, address, key)
+    }
+
+    fn tstore(&mut self, address: Address, key: U256, value: U256) {
+        JournalTr::tstore(&mut self.inner, address, key, value)
+    }
+
+    fn log(&mut self, log: Log) {
+        JournalTr::log(&mut self.inner, log)
+    }
+
+    fn selfdestruct(
+        &mut self,
+        address: Address,
+        target: Address,
+    ) -> Result<StateLoad<SelfDestructResult>, <Self::Database as Database>::Error> {
+        JournalTr::selfdestruct(&mut self.inner, address, target)
+    }
+
+    fn warm_account_and_storage(
+        &mut self,
+        address: Address,
+        storage_keys: impl IntoIterator<Item = U256>,
+    ) -> Result<(), <Self::Database as Database>::Error> {
+        JournalTr::warm_account_and_storage(&mut self.inner, address, storage_keys)
+    }
+
+    fn warm_account(&mut self, address: Address) {
+        JournalTr::warm_account(&mut self.inner, address)
+    }
+
+    fn warm_precompiles(&mut self, addresses: HashSet<Address>) {
+        JournalTr::warm_precompiles(&mut self.inner, addresses)
+    }
+
+    fn precompile_addresses(&self) -> &HashSet<Address> {
+        JournalTr::precompile_addresses(&self.inner)
+    }
+
+    fn set_spec_id(&mut self, spec_id: SpecId) {
+        JournalTr::set_spec_id(&mut self.inner, spec_id)
+    }
+
+    fn touch_account(&mut self, address: Address) {
+        JournalTr::touch_account(&mut self.inner, address)
+    }
+
+    fn transfer(
+        &mut self,
+        from: Address,
+        to: Address,
+        balance: U256,
+    ) -> Result<Option<TransferError>, <Self::Database as Database>::Error> {
+        JournalTr::transfer(&mut self.inner, from, to, balance)
+    }
+
+    fn inc_account_nonce(
+        &mut self,
+        address: Address,
+    ) -> Result<Option<u64>, <Self::Database as Database>::Error> {
+        JournalTr::inc_account_nonce(&mut self.inner, address)
+    }
+
+    fn load_account(
+        &mut self,
+        address: Address,
+    ) -> Result<StateLoad<&mut Account>, <Self::Database as Database>::Error> {
+        JournalTr::load_account(&mut self.inner, address)
+    }
+
+    fn load_account_code(
+        &mut self,
+        address: Address,
+    ) -> Result<StateLoad<&mut Account>, <Self::Database as Database>::Error> {
+        JournalTr::load_account_code(&mut self.inner, address)
+    }
+
+    fn load_account_delegated(
+        &mut self,
+        address: Address,
+    ) -> Result<StateLoad<AccountLoad>, <Self::Database as Database>::Error> {
+        let spec = self.spec_id;
+        let is_eip7702_enabled = spec.is_enabled_in(ScrollSpecId::EUCLID);
```

Comment:
The `ScrollJournal` is added specifically to patch this line from the `Journal` implementation, which used a `SpecId` internally and sets the variable with `spec.is_enabled_in(SpecId::PRAGUE);`. Because Prague isn't enabled for `ScrollSpecId::EUCLID`, the `ScrollJournal` is required.